plugins/events/create.js

```diff
@@ -78,12 +78,30 @@ module.exports = () => ({
                     userFactory.get({ username, scmContext })
                 ]).then(([pipeline, user]) => user.getPermissions(pipeline.scmUri)
                     // Check if user has push access
+                    // eslint-disable-next-line consistent-return
                     .then((permissions) => {
                         if (!permissions.push) {
-                            throw boom.unauthorized(`User ${username} `
-                              + 'does not have push permission for this repo');
+                            const newAdmins = pipeline.admins;
+
+                            delete newAdmins[username];
+                            pipeline.admins = newAdmins;
+
+                            return pipeline.update()
+                                .then(() => {
+                                    throw boom.unauthorized(`User ${username} `
+                                    + 'does not have push permission for this repo');
+                                });
                         }
                     })
+                    // user has good permissions, add the user as an admin
+                    .then(() => {
+                        const newAdmins = pipeline.admins;
+
+                        newAdmins[username] = true;
```

Comment:
This is not array but object, so result is same.
```
{ "foo": true, "bar": true}
```
then add `foo` to admins, the result is
```
{"foo": true, "bar": true}
```

It might be too lengthy to check that the user is already in `admins` or not.

Reply:
I think it's verbose to add and update pipeline every time even if the result is same.

Reply:
Updating DB here is certainly unnecessary. I confirmed that it works locally, and I removed `pipeline.update()`

Reply:
It's just a question. Where do we update admins?

Reply:
https://github.com/sonic-screwdriver-cd/screwdriver/blob/0766d0e75c64ca5d6a3f0d5b0e2d2cde0e047611/plugins/events/create.js#L156
https://github.com/sonic-screwdriver-cd/screwdriver/blob/0766d0e75c64ca5d6a3f0d5b0e2d2cde0e047611/plugins/builds/create.js#L73
https://github.com/sonic-screwdriver-cd/screwdriver/blob/0766d0e75c64ca5d6a3f0d5b0e2d2cde0e047611/plugins/pipelines/sync.js#L72

These lines seems to call `pipeline.update()`

Reply:
I think you are right. I fixed to check `admins` before `update`.